sphinx_gallery/gen_gallery.py

```diff
@@ -340,6 +340,33 @@ def call_memory(func):
     gallery_conf["compress_images"] = compress_images
     gallery_conf["compress_images_args"] = compress_images_args
 
+    # deal with matplotlib_animations
+    animations = gallery_conf["matplotlib_animations"]
+    if not isinstance(animations, (tuple, list)):
+        # Original option may be only a boolean.
+        animations = (animations,)
+    if len(animations) not in (1, 2):
+        raise ConfigError(
+            "Matplotlib animation options must be a single bool or "
+            f"(enabled: bool, format: str), not {animations!r}"
+        )
+    enabled = animations[0]
+    if not isinstance(enabled, bool):  # On/off boolean.
+        raise ConfigError("Animation enabled must be a bool")
+    if len(animations) > 1:  # File format.
+        fmt = animations[1]
+        if fmt is not None:
+            if not isinstance(fmt, str):
+                raise ConfigError("Animation file format must be a string or None")
+            if app is not None:
+                # An explicit format means we place animations externally and require
+                # this `video` extension to embed them into the HTML.
+                app.setup_extension("sphinxcontrib.video")
```

Comment:
I haven't followed this closely but does the user need `sphinxcontrib-video` installed? Do we need to raise error here if not installed?

Reply:
It is currently optional, but I don't have a problem with making it required either.

Reply:
Sorry I mean, would it be useful to give more info in the error message here (e.g., when `matplotlib_animations` config is set to xx, 'sphinxcontrib.video' package is required)? What would the error message be with current code? "..module not found" or..?

Reply:
You should get 
```
sphinx.errors.ExtensionError: Could not import extension sphinxcontrib.video (exception: No module named 'sphinxcontrib.video')
```

Reply:
I think it would be nice to give more info in the error message, e.g., "`matplotlib_animations` has specified file format: xx, this requires the `sphinxcontrib.video` package. 

We all know people don't read docs.

Reply:
OK, done.